pod-operation/src/main.rs

```diff
@@ -1,35 +1,202 @@
 use axum::Server;
+use serde_json::json;
 use socketioxide::{extract::SocketRef, SocketIo};
+use socketioxide::extract::{AckSender, Data};
+
 use tracing::{error, info};
 use tracing_subscriber::FmtSubscriber;
-
-mod components;
+mod state;
 mod demo;
-mod handlers;
-
+mod components;
+use std::{sync::{Arc, Mutex}, thread, time::Duration};
+use state::{State, GLOBAL_STATE};
+use std::collections::HashMap;
 use crate::components::pressure_transducer::PressureTransducer;
-use crate::components::signal_light::SignalLight;
+
+struct StateMachine {
+	state_now: Option<State>,
+    enter_actions: HashMap<State, fn()>, 
+	io: SocketIo,
+}
```

Comment:
Chore: having the `StateMachine` definition and implementation in `main.rs` adds unnecessary clutter here. Could we move this to a separate file (e.g. `fsm.rs`)